added short_id and url_photo

caiosweet opened the pull request:
Hi. I'm not a developer, but I propose this change. I have been using this system in Homeassistant for more than a year, copying your work from the ign_sismologia component.
I'd like to see your work integrated into homeassistant. Thanks for the attention.

[Here are the changes for homeassistant](
https://github.com/caiosweet/Package-Natural-Events/tree/caiosweet-patch-1/config/custom_components/ingv_centro_nazionale_terremoti)

Other types of images and urls
```
      [PGA](http://shakemap.rm.ingv.it/shake/f"{short_id}"/download/pga.jpg)
      [PGV](http://shakemap.rm.ingv.it/shake/f"{short_id}"/download/pgv.jpg)
      [TvMap](http://shakemap.rm.ingv.it/shake/f"{short_id}"/download/tvmap.jpg)
      [TvMap2](http://shakemap.rm.ingv.it/shake/f"{short_id}"/download/tvmap_bare.jpg)
```

caiosweet committed:
Update __init__.py

exxamalte commented:
Thank you very much for your contribution.
Could you please also add unit tests for the additional functionality?

exxamalte commented on georss_ingv_centro_nazionale_terremoti_client/__init__.py:
```diff
@@ -91,3 +95,15 @@ def magnitude(self) -> Optional[float]:
         return self._search_in_title(REGEXP_ATTR_REGION)
+    
+    @property
+    def short_id(self) -> Optional[int]:
```
Should we call this `event_id` because that's how it's called in the source.

caiosweet replied:
Okay, I'll try

exxamalte commented on georss_ingv_centro_nazionale_terremoti_client/__init__.py:
```diff
@@ -9,10 +9,14 @@
     .format(CUSTOM_ATTRIBUTE)
 REGEXP_ATTR_REGION = r'Magnitude\(M.{{0,3}}\) [^ ]+[ ]+-[ ]+(?P<{}>.+)$'\
     .format(CUSTOM_ATTRIBUTE)
+REGEXP_ATTR_SHORT_ID = r'eventId=(?P<{}>\d+)$'\
```
`REGEXP_ATTR_EVENT_ID`

caiosweet committed:
Update __init__.py

Ok, is this correct?
I think I will only do damage with github and your code: 0)

I think I will have to change also in the homeassistant file (geo_location.py)

caiosweet commented:
> Thank you very much for your contribution.
> Could you please also add unit tests for the additional functionality?

I did not understand correctly, you mean automation and images? :0)

I also changed in this file [geo_location.py](https://github.com/caiosweet/Package-Natural-Events/blob/9ba43bd8ac0b537668a784e878814b75d3d362eb/config/custom_components/ingv_centro_nazionale_terremoti/geo_location.py)

## Note.
### Images are available from magnitude 3 and up.

In configuration.yaml i have:
```
geo_location:
  - platform: ingv_centro_nazionale_terremoti
    radius: 1000
    minimum_magnitude: 3
```

```
automation:
  - id: test_natural_events_quake_notifications
    alias: test_natural_events_quake_notifications
    initial_state: true
    trigger:
      - platform: geo_location
        source: 'ingv_centro_nazionale_terremoti'
        zone: zone.geoalert
        event: enter
    condition:
      - condition: template
        value_template: >
          {{((as_timestamp(utcnow())-as_timestamp(trigger.to_state.attributes.publication_date))/3600*60)|int < 60 }}
    action:
      - service: notify.telegram
        data_template:
          title: >-
            Rilevato terremoto. Epicentro: {{ trigger.to_state.attributes.region }}
          message: ""
          data:
            photo:
              - url: >-
                 {{trigger.to_state.attributes.image_url}}
                caption: |
                  Rilevato terremoto. Epicentro: {{ trigger.to_state.attributes.region }}
                  {% set data_utc = trigger.to_state.attributes.publication_date %}
                  Rilevato terremoto di magnitudo: {{trigger.to_state.attributes.magnitude}}
                  a una distanza di {{ trigger.to_state.state }} Km da casa. 
                  Epicentro: {{ trigger.to_state.attributes.region }}
                  {{as_timestamp(data_utc)|timestamp_custom ('Data %d/%m/%Y Ore %H:%M:%S')}}
```

```
binary_sensor:
  - platform: template
    sensors:
      lastquake:
        friendly_name: Evento terremoto
        entity_id: 
          - automation.natural_events_quake_notifications
          - sensor.date
        device_class: vibration
        # availability_template: False
        value_template: >-
          {% set last_date = states.geo_location
            | selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
            | sort(attribute='attributes.publication_date')
            | map(attribute='attributes.publication_date') |list|last %}
          {{ ((as_timestamp(utcnow())-as_timestamp(last_date))/3600) <= 24 if last_date else False }}
        attribute_templates:
          distance: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
             |sort(attribute='attributes.publication_date')|map(attribute='state')|list|last}}
          lat: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.latitude')|list|last}}
          long: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.longitude')|list|last}}
          title: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.title')|list|last}}
          region: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.region')|list|last}}
          magnitude: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.magnitude')|list|last}}
          publication_date: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.publication_date')|list|last}}
          event_id: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.event_id')|list|last}}
          image_url: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.image_url')|list|last}}
          attribution: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.attribution')|list|last}}
          level: >-
            {%set m = states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
              |sort(attribute='attributes.publication_date')|map(attribute='attributes.magnitude')|list|last|default(0)%}
              {% set m = m|float %}
              {%if 0<=m<=1.9%}0{%elif 2<=m<=2.9%}1{%elif 3<=m<=3.9%}2{%elif 4<=m<=5.9%}3{%else%}4{%endif%}
          external_id: >-
            {{states.geo_location|selectattr('attributes.source','eq','ingv_centro_nazionale_terremoti')
            |sort(attribute='attributes.publication_date')|map(attribute='attributes.external_id')|list|last|replace('smi:','')}}
```

Lovelace card
```
  type: vertical-stack
  cards:
#  - type: custom:text-divider-row
#    text:  TERREMOTO
  - type: markdown
    style: >
      ha-card {background: none; border-radius: 0px; box-shadow: none; color: gray;}
      .card-content {padding: 0}
    content: >-
      {% set device_tracker = 'device_tracker.oneplus_5t' %}
      {% set data_utc = state_attr('binary_sensor.lastquake', 'publication_date') %}
      {% set magnitudo = (state_attr('binary_sensor.lastquake', 'magnitude')|float) if not none else '0' %}
      {% set code = {0:'White', 1:'Green', 2:'Yellow', 3:'Orange', 4:'Red'} %}
      {% set color = code[state_attr('binary_sensor.lastquake', 'level')|int] %}
      {% set lat = state_attr('binary_sensor.lastquake', 'lat') %}
      {% set lon = state_attr('binary_sensor.lastquake', 'long') %}
      <font color={{color}}>{{as_timestamp(data_utc)|timestamp_custom ('%H:%M:%S del %d-%m-%Y')}}</font>


      Un terremoto di magnitudo <font color={{color}}> {{magnitudo}} </font> 
      è avvenuto nella zona: [{{state_attr('binary_sensor.lastquake', 'region')}}](https://www.google.com/maps/search/?api=1&query={{state_attr('binary_sensor.lastquake', 'lat')}},{{state_attr('binary_sensor.lastquake', 'long')}})
      a <font color={{color}}> {{state_attr('binary_sensor.lastquake', 'distance')}} </font> km da casa,
      con coordinate geografiche (lat, lon) {{lat}}, {{lon}}. 
      
      Tu ti trovi a {{distance(lat, lon, device_tracker) | round(1) if (lat and lon) else none}} km dall'epicentro.


      {% if magnitudo >= 3|int %}
      [Shakemap]({{state_attr('binary_sensor.lastquake', 'image_url')}}) ~ 
      [PGA](http://shakemap.rm.ingv.it/shake/{{state_attr('binary_sensor.lastquake', 'event_id')}}/download/pga.jpg) ~ 
      [PGV](http://shakemap.rm.ingv.it/shake/{{state_attr('binary_sensor.lastquake', 'event_id')}}/download/pgv.jpg) ~ 
      [TvMap](http://shakemap.rm.ingv.it/shake/{{state_attr('binary_sensor.lastquake', 'event_id')}}/download/tvmap.jpg) ~ 
      [TvMap2](http://shakemap.rm.ingv.it/shake/{{state_attr('binary_sensor.lastquake', 'event_id')}}/download/tvmap_bare.jpg) ~ 
      [HaiSentitoIlTerremoto](http://eventi.haisentitoilterremoto.it/{{state_attr('binary_sensor.lastquake', 'event_id')}}/{{state_attr('binary_sensor.lastquake', 'event_id')}}_mcs.jpg)

      <!-- Scegli il titpo di immagine da visualizzare -->
      <img style="border: 2px solid {{color}};" src="{{state_attr('binary_sensor.lastquake', 'image_url')}}"/>
      <!-- <img style="border: 2px solid {{color}};" src="http://shakemap.rm.ingv.it/shake/{{state_attr('binary_sensor.lastquake', 'event_id')}}/download/pga.jpg"/> -->
      {% endif %}
```

[Video](https://hassiohelp.eu/wp-content/uploads/2019/10/Natural_events.mp4)
[Article](https://hassiohelp.eu/2019/10/06/package-eventi-naturali/)

<br>
<p align="center">
<img src="https://github.com/caiosweet/Package-Natural-Events/blob/caiosweet-patch-1/earthquake.jpg">
<br>

[image](https://github.com/caiosweet/Package-Natural-Events/blob/caiosweet-patch-1/earthquake.jpg)

exxamalte commented:
> I did not understand correctly, you mean automation and images? :0)

I meant unit tests like you can find in the `tests` folder. These test ensure that any change in the original code is automatically tested.
If you've never done that before, that's fine. I'll take a look at adding tests then.

caiosweet commented:
Never done that before. I’m sorry. 😅
